proof_of_concept/ddm_client.py

```diff
@@ -106,7 +208,19 @@ def retrieve_asset(self, site_name: str, asset_id: str
         """Obtains a data item from a store."""
         site = self._get_site('name', site_name)
         if site is not None and site.store is not None:
-            return site.store.retrieve(asset_id, self._party)
+            safe_asset_id = quote(asset_id, safe='')
+            r = requests.get(
+                    f'{site.endpoint}/assets/{safe_asset_id}',
+                    params={'requester': self._party})
+            if r.status_code == 404:
+                raise KeyError('Asset not found')
+            elif not r.ok:
+                raise RuntimeError('Server error when retrieving asset')
```

Comment:
Well, either the asset is found and we return it (HTTP 200), or it hasn't been found (HTTP 404) and we give an error. It may be that the asset doesn't exist yet, but that is still a not found, in the future when I implement proper error handling for workflows there may be a "run failed, won't exist, here's the error message" response, but that doesn't exist yet (and will still be an HTTP 200). So if you get another message then something went wrong unexpectedly, which means there was a server error.

requests.Response.ok is False if the status code is 400 or larger. Since we can only get a 200, 404, or 500, and 404 has already been handled, there must be either a 500 series code signalling a server error, or an unexpected 400 series code, which since it shouldn't have been sent also means there was a server error.

Reply:
Yes, that makes sense.